src/components/Form.test.js

```diff
@@ -305,11 +330,209 @@ describe("Form", () => {
+          {RenderChild}
+        </Form>
+      );
+
+      expect(
+        screen.queryByText("This name is already taken")
+      ).not.toBeInTheDocument();
+      expect(
+        screen.queryByText("Try to spell it differently")
+      ).not.toBeInTheDocument();
+      expect(screen.queryByText("Please enter a name")).toBeInTheDocument();
+      expect(screen.queryByText("Please enter an age")).toBeInTheDocument();
+      expect(
+        screen.queryByText("Please enter a street number")
+      ).not.toBeInTheDocument();
+
+      userEvent.type(screen.getByLabelText("Name"), "Helena");
+
+      await waitFor(() => {
+        expect(
+          screen.queryByText("This name is already taken")
+        ).toBeInTheDocument();
+        expect(
+          screen.queryByText("Try to spell it differently")
+        ).toBeInTheDocument();
+        expect(screen.queryByText("Please enter a name")).not.toBeInTheDocument();
+        expect(screen.queryByText("Please enter an age")).toBeInTheDocument();
+
+        expect(
+          screen.queryByText("Please enter a street number")
+        ).toBeInTheDocument();
+      });
+    });
+
+    it("sets form state to initial values when resetForm is called without argument", async () => {
+      const initialValues = {
+        name: "",
+        address: { streetNumber: "", streetName: "" },
+      };
+      const initialErrors = {
+        name: ["Please enter a name"],
+        address: { streetNumber: ["Please enter a street number"] },
+      };
+
+      const RenderChild = ({ resetForm }) => (
+        <>
+          <Input name="name" label="Name" />
+          <Input name="address.streetNumber" label="Street number" />
+          <Input name="address.streetName" label="Street name" />
+          <Button
+            testId="resetButton"
+            onClick={() => {
+              resetForm();
+            }}
+          >
+            Reset
+          </Button>
+        </>
+      );
+
+      render(
+        <Form initialValues={initialValues} initialErrors={initialErrors}>
+          {RenderChild}
+        </Form>
+      );
+
+      expect(screen.queryByText("Please enter a name")).toBeInTheDocument();
+      expect(
+        screen.queryByText("Please enter a street number")
+      ).toBeInTheDocument();
+      expect(screen.queryByDisplayValue("Helena")).not.toBeInTheDocument();
+      expect(screen.queryByDisplayValue("22")).not.toBeInTheDocument();
+
+      userEvent.type(screen.getByLabelText("Name"), "Helena");
+      userEvent.type(screen.getByLabelText("Street number"), "22");
+
+      await waitFor(() => {
+        expect(
+          screen.queryByText("Please enter a name")
+        ).not.toBeInTheDocument();
+        expect(
+          screen.queryByText("Please enter a street number")
+        ).not.toBeInTheDocument();
+        expect(screen.queryByDisplayValue("Helena")).toBeInTheDocument();
+        expect(screen.queryByDisplayValue("22")).toBeInTheDocument();
+      });
```

Comment:
Again, these `expect`s do not seem to belong to this test. This test is about making sure that `resetForm` does its job. Can we remove these `expect`s? 
I want to make sure that we keep the tests as small as possible, and every test is focused on one thing.

Reply:
Isn't it important to show 

1. initial state
2. state after changes
3. state returns to initial state after resetForm

?

Reply:
This test is about making sure that `resetForm()` is causing the state to be set to the initial state.
So, I expect to see:
1. set up (no expects here)
2. call resetForm()
3. check that state is initial state using expects